src/node_http_parser.cc

```diff
@@ -478,7 +460,144 @@ class Parser : public BaseObject {
   }
 
 
- private:
+  static void Consume(const FunctionCallbackInfo<Value>& args) {
+    Parser* parser = Unwrap<Parser>(args.Holder());
+    Local<External> stream_obj = args[0].As<External>();
```

Comment:
Checking if the object is external from JS isn't possible. I'd feel more comfortable with a `CHECK(args[0]->IsExternal());` just above.